freeipa/src/main/java/com/sequenceiq/freeipa/util/KrbKeytabGenerator.java

```diff
@@ -0,0 +1,54 @@
+package com.sequenceiq.freeipa.util;
+
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.ActorKerberosKey;
+import com.sequenceiq.cloudbreak.common.service.Clock;
+import org.apache.directory.server.kerberos.shared.keytab.KeytabEncoder;
+import org.apache.directory.server.kerberos.shared.keytab.KeytabEntry;
+import org.apache.directory.shared.kerberos.KerberosTime;
+import org.apache.directory.shared.kerberos.codec.types.EncryptionType;
+import org.apache.directory.shared.kerberos.components.EncryptionKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.util.Base64;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public final class KrbKeytabGenerator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KrbKeytabGenerator.class);
+
+    private static final int PRINCIPAL_TYPE_1 = 1;
```

Comment:
What is this principal type?

Reply:
This is goes away in my upcoming change that uses kerby-util package. Internally it will still set a name type of 1 meaning NT-PRINCIPAL, which is what we want I believe.

I'm not sure how important this field is either ways, as per https://tools.ietf.org/html/rfc4120

"The name-type SHOULD be treated only as a hint to interpreting the meaning of a name.  It is not significant when checking for equivalence. Principal names that differ only in the name-type identify the same principal."